bash/users.sh

```diff
@@ -0,0 +1,263 @@
+#          username: name of user to be fetched. Can be left empty
+#                    if user ID is known  $3
+#          user_ID: The ID of a user as saved in the user's profile.
+#                   If the user ID is not known, username can be passed
+#                   and the ID will be found automatically by the function
+#                   id_from_username(headers, url, username) accessible by 
+#                   using $3
+#  Output - Returns a json object containing user information
+get_user(){
+    id=$4
+    if [ -z "$id" ];
+    then
+        id=$(id_from_username "$1" $2 "$3")
+        id=${id:1:-1}
+    fi 
+    url=$2"/"$id
+    response=$(get_helper "$1" $url)
+    user=`$response`
+    echo $user
+}
+
+#  new_user(headers, url, req_data) - This will post a new user to the user list 
+#                                  to the customer instance. User will not have
+#                                  accepted the EULA yet
+#  Input - headers: the headers for the curl request  $1
+#          url: Desired url endpoint  $2
+#          req_data: json object containing relevant new user information accessible 
+#          by using $2
+#  Output - Returns a json object containing created user's data. The OTC key will
+#           need to be pulled in order to accept EULA and set new user password
+new_user(){
+    echo $(post_helper "$1" $2 "$3")
+}
+
+# accept_EULA(headers, url, OTC, email) - Before a new user can set password and
+#                                 start working, the user must accept the 
+#                                 EULA. This requires the OTC returned in the
+#                                 json object returned from new_user(headers, url, req_data)
+# Input - headers: the headers for the curl request  $1
+#         url: Desired url endpoint  $2
+#         OTC: One time token returned in the json object from new_user(headers, url, req_data)  $3
+#         email: email address of new user  $4
+# Output - Returns a token as a string that can be used to login the new user for 
+#          the first time and allows new user to set password
+accept_EULA(){
+    data="{\"token\":\"$3\",\"email\":\"$4\",\"acceptedSeagateTermsAndConditionsAndPrivacyPolicy\":true}"
+    echo $(post_helper "$1" $2 $data)
+}
+
+# first_password_set(headers, url, token, password) - Uses token returned from accepting the EULA
+#                                             to set new user password for the first time
+# Input - headers: the headers for the curl request  $1
+#         url: Desired url endpoint  $2
+#         token: json web token returned fromt the function accept_EULA(headers, url, OTC, email)  $3
+#         password: Desired first password  #4
+# Output - Returns a json object with new user information and a new created password as a key: value pair
+first_password_set(){
+    post_data="{\"token\": \"$3\",\"password\": \"$4\",\"name\": \"New Guy\",\"phone\": \"123-456-7890\",\"location\": \"Longmont, CO\"}"
```

Comment:
Should phone and location also be arguments?

Reply:
This is in relation to a bug that Jessica and I found in which the call to patch for some reason required the full json object. Phone and location were not optional and the patch would fail if they weren't included. I believe that has since been fixed and I didn't communicate that to Dylan

Reply:
This is fixed in the staging instance and has sense been removed from the file